.github/workflows/keepalive.yml

```
@@ -5,17 +5,25 @@ name: Keep-alive (prevent free-tier spin-down)
+#              GET /api/ping (primary) + POST /api/scheduler/tick (secondary)
+#   Layer B: THIS workflow — every 3 min, GET /api/ping + GET /api/health
+#   Layer C: autonomous-cycle.yml — every 2 min, GET /api/ping + /api/autonomy/tick
+#   Layer D: backend BOT_KEEPALIVE — every 10 min, self-ping (last resort)
+#
+# NOTE: For hard 24×7 uptime, also point a free external monitor
+# (UptimeRobot / cron-job.org) at /api/ping on a 5-minute interval.
 
 on:
   schedule:
-    - cron: "*/5 * * * *"
+    - cron: "*/3 * * * *"
```

actionlint (1.7.12)

[error] scheduled job runs too frequently. it runs once per 180 seconds. the shortest interval is once every 5 minutes

(events)